fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -34,32 +27,30 @@ def wait_for_build_processing_to_be_complete(app_id: nil, platform: nil, train_v
 
       private
 
-      def watching_build(app_id: nil, platform: nil)
-        processing_builds = Spaceship::TestFlight::Build.all_processing_builds(app_id: app_id, platform: platform, retry_count: 2)
-
-        watched_build = processing_builds.sort_by(&:upload_date).last
-        watched_build || Spaceship::TestFlight::Build.latest(app_id: app_id, platform: platform)
-      end
-
       def matching_build(watched_train_version: nil, watched_build_version: nil, app_id: nil, platform: nil)
-        matched_builds = Spaceship::TestFlight::Build.builds_for_train(app_id: app_id, platform: platform, train_version: watched_train_version, retry_count: 2)
-        matched_builds.find { |build| build.build_version == watched_build_version }
+        # Get build deliveries (newly uploaded processing builds)
+        client = Spaceship::ConnectAPI::Base.client
+        build_deliveries = client.get_build_deliveries(filter: { app: app_id, cfBundleShortVersionString: watched_train_version, cfBundleVersion: watched_build_version }, limit: 1)
+        build_delivery = build_deliveries.first
+
+        # Get processed builds when no longer in build deliveries
+        unless build_delivery
+          matched_builds = Spaceship::TestFlight::Build.all(app_id: app_id, platform: platform)
```

Comment:
I _think_ I'd like to leave this alone for now until we get our reworked `Spaceship::ConectAPI` module with models in it? I don't want this to stay like this for long but this will need to get reworked in the future anyway so I think I'd prefer to leave it like it is since it "works" 😬😬😬😬😬😬